### ISSUE TITLE
Spec generation gives error for Sentinel

### ISSUE THREAD

**Issue by vidartf** [jupyter-widgets/ipywidgets#2743]
running the generate spec command, I get

```bash
C:\dev\ipywidgets [datetime ↑1]> python ./packages/schema/generate-spec.py -f json-pretty > packages/schema/jupyterwidgetmodels.latest.json
Traceback (most recent call last):
  File "./packages/schema/generate-spec.py", line 208, in <module>
    indent=2, separators=(',', ': ')))
  File "C:\Miniconda3\lib\json\__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "C:\Miniconda3\lib\json\encoder.py", line 201, in encode
    chunks = list(chunks)
  File "C:\Miniconda3\lib\json\encoder.py", line 429, in _iterencode
    yield from _iterencode_list(o, _current_indent_level)
  File "C:\Miniconda3\lib\json\encoder.py", line 325, in _iterencode_list
    yield from chunks
  File "C:\Miniconda3\lib\json\encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "C:\Miniconda3\lib\json\encoder.py", line 325, in _iterencode_list
    yield from chunks
  File "C:\Miniconda3\lib\json\encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "C:\Miniconda3\lib\json\encoder.py", line 438, in _iterencode
    o = _default(o)
  File "C:\Miniconda3\lib\json\encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type Sentinel is not JSON serializable
```

**Comment by vidartf** [comment]
Note this is on my branch for #2715, not on master

**Comment by vidartf** [comment]
After some debugging, it seems the generation code struggles to find the default value for this specification:
```python
Union((CInt(1), Enum("any")))
```

**Comment by vidartf** [comment]
There were some other issues as well with unions, fix propsed in #2548.

**Comment by jasongrout** [comment]
> There were some other issues as well with unions, fix proposed in #2715.

Setting as 8.0, then.

**Comment by marimeireles** [comment]
hey @vidartf, I was trying to have a look at this, but I can't reproduce your error...
Building `ipywidgets` with the #2715 PR leads to the following error for me:
```
(ipywidgets) mariana@wintermute ~/development/ipywidgets (datetime) $ python ./packages/schema/generate-spec.py -f json-pretty > packages/schema/jupyterwidgetmodels.latest.json
Traceback (most recent call last):
  File "/home/mariana/development/ipywidgets/./packages/schema/generate-spec.py", line 232, in <module>
    spec = create_spec(widgets_to_document)
  File "/home/mariana/development/ipywidgets/./packages/schema/generate-spec.py", line 213, in create_spec
    widget_data.append(jsonify(widget_name, widget, widget_list))
  File "/home/mariana/development/ipywidgets/./packages/schema/generate-spec.py", line 193, in jsonify
    default=jsdefault(trait)
  File "/home/mariana/development/ipywidgets/./packages/schema/generate-spec.py", line 116, in jsdefault
    default = trait.make_dynamic_default()
AttributeError: 'Union' object has no attribute 'make_dynamic_default'
```

**Comment by vidartf** [comment]
@marimeireles I think that code was changed with traitlets 5, good catch.

**Comment by marimeireles** [comment]
Is the issue still valid @vidartf ?

**Comment by vidartf** [comment]
Closed in #2715.
